src/site/content/en/blog/hands-on-portals/index.md

```diff
@@ -0,0 +1,203 @@
+ - [The portal host interface](https://wicg.github.io/portals/#the-portalhost-interface) … Adds a `portalHost` object to the `window` object. This lets you check if the page is embedded as a portal element. It also provides an interface for messaging (`postMessage`) back to the host.
+ - [The PortalActivateEvent interface](https://wicg.github.io/portals/#the-portalactivateevent-interface) … An event that fires when the portal is activated. There is a neat function called `adoptPredecessor` which you can use to retrieve the previous page as a `<portal>` element. This allows you to create seamless navigations and composed experiences between two pages.
+
+Let’s look beyond the basic usage pattern. Here is a non-exhaustive list of what you can achieve with Portals along with sample code.
+### Customize the style when embedded as a portal element
+```javascript
+// Detect whether this page is hosted in a portal
+if (window.portalHost) {
+  // Customize the UI when being embedded as a portal
+}
+```
+### Messaging between the portal element and portal host
+```javascript
+// Send message to the portal element
+const portal = document.querySelector('portal');
+portal.postMessage({someKey: someValue}, ORIGIN);
+
+// Receive message via window.portalHost
+window.portalHost.addEventListener('message', evt => {
+  const data = evt.data.someKey;
+  // handle the event
+});
+```
+### Activating the portal element and receiving the `portalactivate` event
+```javascript
+// You can optionally add data to the argument of the activate function
+portal.activate({data: {'somekey': 'somevalue'}});
+
+// The portal content will receive the portalactivate event
+// when the activate happens
+window.addEventListener('portalactivate', evt => {
+  // Data available as evt.data
+  const data = evt.data;   
+});
+```
+### Retrieving the predecessor
+```javascript
+// Listen to the portalactivate event
+window.addEventListener('portalactivate', evt => {
+  // ... and creatively use the predecessor
+  const portal = evt.adoptPredecessor();
+  document.querySelector('someElm').appendChild(portal);
+});
+```
+### Knowing your page was adopted as a predecessor
+```javascript
+// The activate function returns a Promise.
+// When the promise resolves, it means that the portal has been activated.
+// If this document was adopted by it, then window.portalHost will exist.
+portal.activate().then(_ => {
+  // Check if this document was adopted into a portal element.
+  if (window.portalHost) {
+    // You can start communicating with the portal element
+    // i.e. listen to messages
+    window.portalHost.addEventListener('message', evt => {
+      // handle the event
+    });
+  }
+});
+```
+By combining all the features supported by Portals, you can build really fancy user experiences. For instance the demo below demonstrates how Portals can enable a seamless user experience between a website and a third party embed content. 
```

Comment:
Small nit, missing comma after 'For instance'.

Reply:
Thank you!